libqtile/backend/wayland/drawer.py

```diff
@@ -64,25 +54,22 @@ def _draw(
         if height > self._win.height - offsety:
             height = self._win.height - offsety
 
-        # Paint RecordingSurface operations our window's ImageSurface
-        with cairocffi.Context(self._source) as context:
-            context.set_source_surface(self.surface)
-            context.paint()
-
-        # Copy drawn ImageSurface data into rendered wlr_texture
-        self._win.texture.write_pixels(
-            self._stride,
-            width,
-            height,
-            cairocffi.cairo.cairo_image_surface_get_data(self._source._pointer),
-            dst_x=offsetx,
-            dst_y=offsety,
-        )
-        self._win.damage()
+        # Paint recorded operations to our window's underlying ImageSurface
+        with cairocffi.Context(self._win.surface) as context:
+            context.set_operator(cairocffi.OPERATOR_SOURCE)
```

Comment:
do we still want to remove this line?

Reply:
I made a separate PR to rework the `Drawer` class. Ideally, that should be rebased onto this.

Reply:
@jwijenbergh that operator is actually really important. Was this the line that caused issues for you when testing this branch? Might be worth you testing the drawer refactor once it's in a slightly better state.